### DISCUSSION TITLE
Q: frame-exact (consistent) timing of lavfi audio waveform?

### DISCUSSION THREAD

**Discussion by pjotrek-b** [mpv-player/mpv#14707] [Q&A]
I'm using ffmpeg's lavfi in mpv to display the audio waveforms:

mpv --log-file=audio_waveform.log ~/Video/dva_fidelity2-90s.avi --lavfi-complex='[aid1] asplit [ao] [a1] ; [a1] showwaves=size=600x200:rate=25:mode=line:colors=white:scale=sqrt:split_channels=1 [vida] ; [vid1] [vida] overlay=(W-w)/2:H-h [vo]' --pause

I'd like to show the waveform of the *exact* frame being shown.
My reference video is i-frame-only FFV1, PCM (so I don't expect any lossy-or-GOP related fuzzyness) :wink: 

In my screenshots, you can see the very same frame - with 2 different positions/offsets of the same audio waveform shown:
![Screenshot_2024-02-06_11-02-21](https://github.com/mpv-player/mpv/assets/2419195/3df80ec8-591a-4640-95c7-b02152511a17)
![Screenshot_2024-02-06_11-08-17](https://github.com/mpv-player/mpv/assets/2419195/714413d2-9927-4781-87a1-a76e77393d17)

If I move back/forth (eg seek by-frame using `,`and `.` keys) - the shown waveform is rendered with a changing offset, depending on how (much) I went back or forth.

**Is there anything I can do to force mpv to show *the same audio samples* at the same time/position consistently?**

Thank you very much in advance :smile:

**Comment by pjotrek-b** [comment]
I think the reason is, that the "showwaves" filter doesn't scale correctly to show only "the audio for the currently shown video frame", but rather increases the FPS value (to about 44.444 fps for a 25fps/48kHz source).

Is there any (other/better/consistent) way to have MPV show the audio waveform for the current frame/image?

**Comment by pjotrek-b** [comment]
@kasper93: Closed as completed?
Awesome! Thank you.
What has been changed?

**Comment by kasper93** [comment]
> What has been changed?

The issue tracker is no longer for questions or discussions.
